cluster/src/main/java/io/scalecube/cluster/Cluster.java

```diff
@@ -178,14 +176,8 @@ public static ICluster joinAwait(ClusterConfig config) {
         failureDetector.start();
         gossip.start();
         return membership.start();
-      });
-
-    return transform(clusterFuture, new Function<Void, ICluster>() {
-      @Override
-      public ICluster apply(@Nullable Void param) {
-        return Cluster.this;
-      }
-    });
+      }, MoreExecutors.directExecutor());
+    return clusterFuture.thenApplyAsync(aVoid -> Cluster.this, MoreExecutors.directExecutor());
```

Comment:
I was thinking what if we do it in a more Java 8 like functional style to put instead of `MoreExecutors.directExecutor()` just `Runnable::run`, will it work?